### PR TITLE
Add internal battery measurement example

### PR DESCRIPTION
**Pull request by mfalkvidd**
For an update to https://www.mysensors.org/build/battery#measuring-and-reporting-battery-level based on work by skywatch in https://forum.mysensors.org/topic/11555/measuring-battery-voltage-which-way-is-best/

### CODE REVIEW

**Comment by Yveaux** [comment]
I would suggest to use an external library (e.g. https://github.com/Yveaux/Arduino_Vcc) instaed.

**Comment by mfalkvidd** [comment]
Do we have a good strategy for handling external dependencies?
Move this example to the MySensorsArduinoExamples repo, and copy the current version of your library into MySensorsArduinoExamples? That’s what we’ve done so far, but it isn’t a very good strategy imo.

**Comment by Yveaux** [comment]
External libraries in Arduino are a real pain. PlatformIO handles it a lot better, allowing automatic installation of libraries, direct referencing and pinning specific versions. This is however a separate discussion...
In line with other examples this would require a copy of the Vcc library indeed.
Later we can decide how to improve on this in general.

**Comment by mfalkvidd** [comment]
How about using `hwCPUVoltage()`? It seems to do something similar.

**Comment by user2684** [comment]
For what it matters, on the NodeManager side I'm also using hwCPUVoltage()  (https://github.com/mysensors/NodeManager/blob/master/sensors/SensorBattery.h)

**Comment by mfalkvidd** [comment]
Any additional comments before I merge?

**Comment by Yveaux** [comment]
I suppose you tested it? If so, go ahead 👍🏻

File: examples/BatteryPoweredSensorInternal/BatteryPoweredSensorInternal.ino [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+{
+  // Send the sketch version information to the gateway and Controller
+  sendSketchInfo("Battery Meter", "1.0");
+}
+
+void loop()
+{
+  // get the battery Voltage
+  long batteryMillivolts = hwCPUVoltage();
+  int batteryPcnt = batteryMillivolts / FULL_BATTERY / 1000.0 * 100 + 0.5;
```
Comment (tekka007) [inline examples/BatteryPoweredSensorInternal/BatteryPoweredSensorInternal.ino:59]: Why adding 0.5?

Comment (Yveaux) [reply examples/BatteryPoweredSensorInternal/BatteryPoweredSensorInternal.ino:59]: Round to next full int value

Comment (tekka007) [reply examples/BatteryPoweredSensorInternal/BatteryPoweredSensorInternal.ino:59]: i.e. full battery = 101%

Comment (Yveaux) [reply examples/BatteryPoweredSensorInternal/BatteryPoweredSensorInternal.ino:59]: Only if the calculation adds up to a value of 100.5 or more..
No need to blame the +0.5 😉

**Comment by mfalkvidd** [comment]
> I suppose you tested it? If so, go ahead 👍🏻

I've tested with a 5V Pro Mini as serial gateway. It works fine:

```
16:25:59.014 -> Battery voltage: 4.48V
16:25:59.014 -> Battery percent: 149 %
```

I also tested with a nodemcu. It returns 0V, as expected since `hwCPUVoltage()` is not supported.

**Comment by Yveaux** [comment]
```
16:25:59.014 -> Battery voltage: 4.48V
16:25:59.014 -> Battery percent: 149 %
```
That 149% is a bit funky, but that's likely because you didn't tune it or ran the pro mini from 3V, I suppose?

**Comment by mfalkvidd** [comment]
> That 149% is a bit funky, but that's likely because you didn't tune it or ran the pro mini from 3V, I suppose?

I ran the example as-is, with `#define FULL_BATTERY 3`

**Comment by Yveaux** [comment]
Then it makes total sense! 👍🏻

**Comment by mfalkvidd** [comment]
On release: Change the battery page to point to the master branch instead of the development branch.

**Pull request merged**